src/utils.ts

```diff
@@ -46,17 +52,26 @@ class ReactElementTypeError extends Error {
   }
 }
 
-function checkReactElement(element, ...args) {
-  if (!VALID_ELEMENTS.includes(element.type) && !element._fiber) {
-    throw new ReactElementTypeError(element, ...args);
+function checkReactElement(
+  element: ReactTestInstance | null,
+  matcherFn: Function,
+  context: jest.MatcherContext,
+) {
+  if (!element) {
+    throw new ReactElementTypeError(element, matcherFn, context);
+  }
+
+  // @ts-expect-error not sure where to get this fiber property
+  if (!element._fiber && !VALID_ELEMENTS.includes(element.type.toString())) {
+    throw new ReactElementTypeError(element, matcherFn, context);
   }
 }
 
-function getType({ type }) {
+function getType({ type }: any) {
```

Comment:
Try `ReactElement` and/or `JSX.Element`

Reply:
Used type directly from the ReactTestInstance and removed this helper